index.js

```diff
@@ -27,28 +26,23 @@ function MultiDrive (name, opts, cb) {
   assert.equal(typeof opts, 'object', 'multidrive: opts should be an object')
   assert.equal(typeof cb, 'function', 'multidrive: cb should be a function')
 
-  var self = this
-
-  this.db = namedLevel(name)
-  this.opts = opts
-  this.archives = {}
-  this.queue = []
+  var db = namedLevel(name)
 
-  var rs = this.db.createReadStream()
+  var rs = db.createReadStream()
   pump(rs, concat({ encoding: 'json' }, sink), (err) => {
-    if (err) return cb(err)
-    cb(null, self)
+    if (err) cb(err)
   })
 
   function sink (pairs) {
-    mapLimit(pairs, 1, iterator, function (err) {
+    mapLimit(pairs, 1, iterator, function (err, archivesArray) {
       if (err) return cb(err)
-      while (self.queue.length) self.queue.shift()()
+      var archivesMap = keyBy(archivesArray, archive => archive.metadata.directory)
```

Comment:
given this is library code that's meant to be compatible with `hyperdrive` and `hypercore`, could we keep this completely ES5? - thanks!